prov/efa/src/efa_cntr.c

```diff
@@ -159,11 +182,17 @@ int efa_cntr_open(struct fid_domain *domain, struct fi_cntr_attr *attr,
 	if (!cntr)
 		return -FI_ENOMEM;
 
+	dlist_init(&cntr->ibv_cq_poll_list);
 	efa_domain = container_of(domain, struct efa_domain,
 				  util_domain.domain_fid);
 
-	ret = ofi_cntr_init(&efa_prov, domain, attr, &cntr->util_cntr,
-			    &ofi_cntr_progress, context);
+	if (efa_domain->info->ep_attr->type == FI_EP_RDM)
+		ret = ofi_cntr_init(&efa_prov, domain, attr, &cntr->util_cntr,
+				    &efa_rdm_cntr_progress, context);
+	else
+		ret = ofi_cntr_init(&efa_prov, domain, attr, &cntr->util_cntr,
+				    &ofi_cntr_progress, context);
+
```

Comment:
nit for DRY/maintainability (plus there's an existing `typedef`):

```suggestion
	ofi_cntr_progress_func cntr_progress_func = efa_domain->info->ep_attr->type == FI_EP_RDM
		? efa_rdm_cntr_progress
		: ofi_cntr_progress;

	ret = ofi_cntr_init(&efa_prov, domain, attr, &cntr->util_cntr,
			    &cntr_progress_func, context);

```

Reply:
Thanks! updated.